### PR TITLE
Update the design doc for system-wide profiling changes

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -1,45 +1,29 @@
...
 
-Parca Agent uses BPF CO-RE (Compile Once – Run Everywhere) using [libbpf](https://github.com/libbpf/libbpf), and pre-compiles all BPF programs, and statically embeds them in the target binary, from where it is loaded via libbpf when used. This means that Parca Agent does not need to compile the BPF program at startup or runtime like when using [bcc-tools](https://github.com/iovisor/bcc/tree/master/tools), meaning no Clang & LLVM, nor kernel headers need to be installed on the host. The only requirement is a [BTF](https://www.kernel.org/doc/html/latest/bpf/btf.html) capable Kernel (Linux Kernel 4.18+).
+Parca Agent is a whole-system profiler. It collects stack traces from all the processes that run on the host system. This provides more insightful data about all the aspects of the system to the user. Please see our [blog post](https://www.polarsignals.com/blog/posts/2022/08/24/system-wide-profiling/) about internals of this mechanism.
```
Comment (javierhonduco) [inline docs/design.md:5]: nit: "more insights" sounds a bit better to me

File: docs/design.md [hunk 2]
```diff
@@ -1,45 +1,29 @@
...
-### systemd
-
-systemd starts all services in a cgroup, and conveniently and consistently mounts the cgroup hierarchy. Unfortunately [`perf_event` is one of the few cgroups systemd does not automatically manage](https://systemd.io/CGROUP_DELEGATION/#controller-support). This means the `perf_event` cgroup needs to be separately managed by Parca Agent. It will replicate the cgroup hierarchy under `system.slice`, into `perf_event` and sync the `cgroup.procs` once per second.
-
-Labels provided by the systemd discovery are: `node` and `systemd_unit`.
-
 ## Obtaining raw data
 
-Parca Agent obtains the raw data by attaching an BPF program to a [Linux cgroup](https://en.wikipedia.org/wiki/Cgroups) using [perf_event_open](https://man7.org/linux/man-pages/man2/perf_event_open.2.html). It instructs the Kernel to call the BPF program every 100 times per second.
+Parca Agent obtains the raw data by attaching an eBPF program to a process using [perf_event_open](https://man7.org/linux/man-pages/man2/perf_event_open.2.html). It instructs the Kernel to call the BPF program every 100 times per second.
```
Comment (javierhonduco) [inline docs/design.md:21]: I don't think this is correct, we don't attach to processes but to perf events. Perhaps we can mention that weuse the `cycles` event?

File: docs/design.md [hunk 3]
```diff
@@ -1,45 +1,29 @@
...
 
 The way BPF programs communicate with user-space uses BPF maps. The Parca Agent BPF program records data in two maps:
 
 * **Stack traces**: The stack traces map is made up of the stack trace ID as the key and the memory addresses that represent the code executed that represents that stack trace.
-* **Counts**: The counts map is made up of a key that is a triple of PID, user-space stack ID, and kernel-space stack ID and value is the amount of times that stack trace ID has been observed.
+* **Counts**: The counts map is made up of a key that is a triple of PID, user-space stack ID, and kernel-space stack ID and value is the number of times that stack trace ID has been observed.
```
Comment (javierhonduco) [inline docs/design.md:26]: Nice!! ✨

**Review by javierhonduco — APPROVED** [review]
Great changes! Thanks for doing them!!

File: docs/design.md (hunk 2 above)
Comment (kakkoyun) [reply docs/design.md:21]: Do you have link to put in here?

**Review by javierhonduco — APPROVED** [review]

**Pull request merged**